src/migrations/tasks/20230315112325_service-areas-table.ts

```diff
@@ -0,0 +1,107 @@
+import { makeDomainLogger } from "back-end/lib/logger";
+import { console as consoleAdapter } from "back-end/lib/logger/adapters";
+import Knex from "knex";
+
+const logger = makeDomainLogger(consoleAdapter, "migrations");
+
+enum TWUServiceArea {
+  Developer = "DEVELOPER",
+  DataSpecialist = "DATA_SPECIALIST",
+  ScrumMaster = "SCRUM_MASTER",
+  DevopsSpecialist = "DEVOPS_SPECIALIST"
+}
+
+export async function up(connection: Knex): Promise<void> {
+  // Service Areas
+  await connection.schema.createTable("serviceAreas", (table) => {
+    table.increments("id").primary().unique().notNullable();
+    table.enu("serviceArea", Object.values(TWUServiceArea)).notNullable();
+    table.string("name").notNullable();
+  });
+  Object.keys(TWUServiceArea).forEach(async (key) => {
+    await connection("serviceAreas").insert({
+      serviceArea: TWUServiceArea[key as keyof typeof TWUServiceArea],
+      name: key
+    });
+  });
+  logger.info("Created serviceAreas table");
+
+  // Organization Service Areas
+  await connection.schema.createTable(
+    "twuOrganizationServiceAreas",
+    (table) => {
+      table.integer("serviceArea").references("id").inTable("serviceAreas");
+      table.uuid("organization").references("id").inTable("organizations");
+      table.primary(["serviceArea", "organization"]);
+    }
+  );
+  logger.info("Created twuOrganizationServiceAreas table");
+
+  await connection.schema.createTable("twuResources", (table) => {
+    table.uuid("id").primary().unique().notNullable();
+    table.integer("serviceArea").references("id").inTable("serviceAreas");
+    table
+      .uuid("opportunityVersion")
+      .references("id")
+      .inTable("twuOpportunityVersions");
```

Comment:
I notice opportunities fail on deletion now and wonder if we're missing a delete on cascade somewhere here? I wonder also if the "id" column is necessary? For context, look at how `twuResourceQuestions` is currently doing something similar. Something like the following might work, for instance: 
```  
 table.uuid("opportunityVersion")
      .references("id")
      .inTable("twuOpportunityVersions")
      .notNullable()
      .onDelete("CASCADE");
      table.primary(["opportunityVersion"]);
```